### DISCUSSION TITLE
TCMenu 4.1 downgrade to 4.0

### DISCUSSION THREAD

**Discussion by tomelgato** [davetcc/tcMenu#399] [General]
On OSX & Pio:

I have a little messup with a project which i updated to 4.1 tcmenu... To nail down the problem i tried to compile it again with 4.0 libs. On project side its no problem to use the old libraries but i struggle on the tcmenu side.
When i use the TCMenu4.0 version i can load the .emf but when exporting the source it always displays 4.1.1 libraries. Why?

The 4.1 tcmenu project is not compiling with 4.x libraries, especially the crystal liquid throws errors, so i need to downgrade:

menu.cpp & menu.h:
cannot declare variable 'renderer' to be of abstract type 'LiquidCrystalRenderer'

tcMenuLiquidCrystal.h:
'void LiquidCrystalRenderer::drawMenuItem(tcgfx::GridPositionRowCacheEntry*, tcgfx::Coord, tcgfx::Coord, const int&)' marked 'override', but does not override
'DrawingFlags' does not name a type

**Comment by tomelgato** [comment]
Solved: ~/.tcmenu/plugins

**Reply by davetcc** [reply]
See the issue, a simple fix will be in 4.2
